src/util.cc

```diff
@@ -825,6 +825,36 @@ double GetLoadAverage() {
 }
 #else
 double GetLoadAverage() {
+# if defined(__linux__)
+  // Only try to open the file once.
+  static int fd = -1;
+  if (fd == -1) {
+    fd = open("/proc/loadavg", O_RDONLY);
+    if (fd < 0) {
+      fd = -2;
+    }
```

Comment:
Why set this if it is still negative and thus ignored by the `fd >=0` check?

Reply:
"fd" is retained across calls.  Setting it to -2 means we already tried to open() it once and failed for whatever reason -- so we should fall back to  the original behavior.